client/src/pages/Interview.js

```diff
@@ -1,88 +1,128 @@
-import React, { useState, useEffect } from 'react';
+import React, { useContext, useState, useEffect, useCallback } from 'react';
 import { useStyles } from '../themes/theme';
+import { useParams } from 'react-router';
 import Grid from '@material-ui/core/Grid';
 import InterviewQuestionDetails from '../components/InterviewQuestionDetails';
 import TextEditor from '../components/TextEditor';
 import InterviewHeader from '../components/InterviewHeader';
 import OutputConsole from '../components/OutputConsole';
 import axios from 'axios';
-import { getInterview, getQuestion } from '../utils/apiFunctions'
+import SocketContext from '../context/socket';
+import { getInterview, getQuestion } from '../utils/apiFunctions';
 import { useHistory } from 'react-router';
 
-const Interview = () => {
+const Interview = ({ userData }) => {
   const classes = useStyles();
+  const socket = useContext(SocketContext);
+  const handleCodeSnippetChange = (codeSnippet) => {
+    setCode(codeSnippet);
+  };
+  const [language, setLanguage] = useState('javascript');
+  const [code, setCode] = useState('');
+  const [value, setValue] = useState('');
+  const [codeResult, setCodeResult] = useState('');
   const history = useHistory();
-  const roomId = history.location.pathname.split('/')[2];
+  // const roomId = history.location.pathname.split('/')[2];
+  const { id: roomId } = useParams();
+
+  useEffect(() => {
+    socket.emit('change_text', code);
+  }, [code, socket]);
 
-  const [codeData, setCodeData] = useState({
-    language: 'javascript',
-    code: '//write your code here',
+  useEffect(() => {
+    socket.on('new_content', (data) => setValue(data));
   });
 
-  const [codeResult, setCodeResult] = useState('');
+  useEffect(() => {
+    socket.emit('code_result', codeResult);
+  }, [codeResult, socket]);
+
   const [pageData, setPageData] = useState({
     isLoaded: false,
     questions: {
       questionOne: null,
-      questionTwo: null
-    }
+      questionTwo: null,
+    },
   });
 
-  const fetchQuestions = async () => {
+  const fetchQuestions = useCallback(async () => {
```

Comment:
probably should be called fetchInterview

Reply:
Fixed